cirq-core/cirq/ops/pauli_string.py

```diff
@@ -271,9 +271,7 @@ def __mul__(self, other: 'cirq.Operation') -> 'cirq.PauliString[Union[TKey, cirq
         pass
 
     @overload
-    def __mul__(
-        self, other: Union[complex, int, float, numbers.Number]
-    ) -> 'cirq.PauliString[TKey]':
+    def __mul__(self, other: Union[complex, numbers.Number]) -> 'cirq.PauliString[TKey]':
```

Comment:
I am getting True for `issubclass(complex, numbers.Number)` -

```suggestion
    def __mul__(self, other: numbers.Number) -> 'cirq.PauliString[TKey]':
```

Reply:
Here also I went the opposite direction. `numbers` types are supposed to be for `instanceof`, but don't work well with mypy. `SupportsComplex` is the proper way to do it with mypy, but `SupportsComplex` only means it has `__complex__` on it, which means you've got to call `complex(other)` before you can safely use it; code and time overhead. The only thing it improves is that mypy won't complain when using `np.int64` or other tangential types, though they still work at run time regardless. So, just leaving this as `complex`.